synedrion/src/uint.rs

```diff
@@ -4,12 +4,155 @@ mod traits;
 
 pub(crate) use crypto_bigint::{
     modular::Retrieve, subtle, CheckedAdd, CheckedMul, CheckedSub, Encoding, Integer, Invert,
-    NonZero, PowBoundedExp, RandomMod, Zero, U1024, U2048, U4096, U512, U8192,
+    NonZero, PowBoundedExp, RandomMod, ShlVartime, WrappingSub, Zero, U1024, U2048, U4096, U512,
+    U8192,
 };
 pub(crate) use crypto_primes::RandomPrimeWithRng;
 
 pub(crate) use bounded::Bounded;
 pub(crate) use signed::Signed;
-pub(crate) use traits::{
-    upcast_uint, HasWide, U1024Mod, U2048Mod, U4096Mod, U512Mod, UintLike, UintModLike,
+pub(crate) use traits::{upcast_uint, HasWide, ToMod, U1024Mod, U2048Mod, U4096Mod, U512Mod};
+
+/// Math functions for generic integers (mostly `Signed`).
+use crypto_bigint::{
+    subtle::{ConditionallySelectable, CtOption},
+    Square,
 };
+use digest::XofReader;
+
+/// Build a `T` integer from an extendable Reader function
+pub(crate) fn uint_from_xof<T>(reader: &mut impl XofReader, modulus: &NonZero<T>) -> T
```

Comment:
A minor thing, but I would prefer the "node" files to only contain re-exports - makes it easier to navigate. Perhaps we can move the `pow` functions to `uint/pow.rs`, and `uint_from_xof` to `tools/hashing.rs`?